core/ordersync/ordersync.go

```diff
@@ -340,6 +340,29 @@ func (s *Service) GetOrders(ctx context.Context, minPeers int) error {
 	return nil
 }
 
+// PeriodicallyGetOrders periodically calls GetOrders. It waits a minimum of
+// minDelay between each call. It will block until there is a critical error or
+// the given context is canceled.
+func (s *Service) PeriodicallyGetOrders(ctx context.Context, minPeers int, minDelay time.Duration) error {
```

Comment:
How hard would it be to add a test for this? FWIW this is pretty simple, so I don't think it's absolutely necessary, but if it's easy, I'd be in favor of adding it to prevent regressions.

Reply:
Talked to Fabio about this too. I think the code in question is simple enough that our time would be better spent elsewhere. However, I did add a test for the new `calculateDelayWithJitter` since it's a little less straightforward.

Reply:
I think that's a fair assessment. Thanks for adding the other test.